src/Camera/Camera.cpp

```diff
@@ -98,11 +98,8 @@ Camera& Camera::SetProjectionMatrixPerspective(float xFov, float aspect, float n
 {
 	_xFov = glm::radians(xFov);
 	const float yFov = (glm::atan(glm::tan(_xFov / 2.0f)) / aspect) * 2.0f;
-	const float h = 1.0f / glm::tan(yFov * 0.5f);
-	const float w = h / aspect;
-	const float a = nearClip / (farClip - nearClip);
-	const float b = (nearClip * farClip) / (farClip - nearClip);
-	_projectionMatrix = glm::mat4x4(w, 0.f, 0.f, 0.f, 0.f, h, 0.f, 0.f, 0.f, 0.f, a, 1.f, 0.f, 0.f, b, 0.f);
+	_projectionMatrix = glm::mat4x4(1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, -1.0, 0.0, 0.0, 0.0, 1.0, 1.0);
+	_projectionMatrix *= glm::perspective(yFov, aspect, nearClip, farClip);
```

Comment:
Do these 2 lines match any of the perspective functions here?
https://github.com/g-truc/glm/blob/45008b225e28eb700fa0f7d3ff69b7c1db94fadf/glm/ext/matrix_clip_space.hpp#L215

I believe there's a define that lets you select the right kind of perspective

Reply:
Afaik you can only choose whether you use OpenGL or Direct3D depth definitions (via define or using the _ZO or _NO functions respectively) the rest is choice between Left handed or right handed coordinates and what parameter to build the perspective with (fov / viewport). I don't think there is any built-in way to make a perspective matrix with a reversed z in glm.

Reply:
I'm trying to better understand the matrix here. I have reduced it to this. Note the inversing of far and near...
I don't quite understand the need for translation though...
```suggestion
	_projectionMatrix = glm::translate(glm::vec3(0.0f, 0.0f, 1.0f)) * glm::perspective(yFov, aspect, farClip, nearClip);
```

Reply:
Because if you just reverse the far and nearPlane you remap the depth between -1 and 0.